src/Simulators/schedule-callback-args.js

```diff
@@ -0,0 +1,160 @@
+ */
+
+/**
+ * A handle for a set of running timers.
+ */
+class TimerSchedule {
+    constructor() {
+        this._timers = [];
+        this.cancel = this.cancel.bind(this);
+    }
+
+    /**
+     * Add a timer to the internal list.
+     * @param {Number} timer Timer descriptor (a number returned from setTimeout)
+     */
+    add(timer) {
+        this._timers.push(timer);
+    }
+    /**
+     * Cancel all upcoming timers inside this handler. Can only be called once.
+     * @returns undefined;
+     */
+    cancel() {
+        this._timers.forEach(clearTimeout);
+        this._timers = null;
+        this.cancel = () => {
+            throw Error(
+                `Tried to call cancel on an already canceled scheduler`
+            );
+        };
+    }
+}
+
+/**
+ * Validate one of these update events.
+ * @private
+ */
+const badUpdateEvent = update =>
+    isNaN(update.elapsed) ||
+    (typeof update.args !== 'function' && !Array.isArray(update.args));
+
+/**
+ * Describe how and when to call a function over time, using an array of
+ * configuration objects.
+ *
+ * Use to test the behavior of asynchronous listeners and components.
+ *
+ * Returns an object with a `.cancel()` method, which when called will
+ * cancel any updates not yet executed.
+ *
+ * @param {UpdateEvent[]} calls List of { elapsed: number, args: any[] }
+ *    objects describing the timing and arguments for executing the callback.
+ * @param {Function} callback Callback to be invoked on the schedule, with the
+ *    configured arguments.
+ * @param {Function} errorHandler Callback to be invoked with any errors that
+ * occur during delayed calls. Defaults to throwing the exception, and async
+ * exceptions are hard to catch.
+ *
+ * @returns {TimerSchedule}
+ */
+export default function scheduleCallbackArgs(
+    calls,
+    callback,
+    errorHandler = defaultErrorHandler
+) {
+    if (
+        !Array.isArray(calls) ||
+        calls.length < 1 ||
+        calls.some(badUpdateEvent)
+    ) {
+        return errorHandler(
+            Error(
+                'First argument must be an array of 1 or more { elapsed: number, args: any[] } objects.'
+            )
+        );
+    }
+
+    if (typeof callback !== 'function') {
+        return errorHandler(
+            Error('Must provide a callback as the second argument.')
+        );
+    }
+
+    const schedule = new TimerSchedule();
+
+    const invoker = getArgs => () => {
+        let args;
+        try {
+            args = getArgs();
+        } catch (getArgsError) {
+            errorHandler(getArgsError);
+            return;
+        }
+        if (Array.isArray(args)) {
+            try {
+                return callback(...args);
+            } catch (callbackError) {
+                errorHandler(callbackError);
+                return;
+            }
+        } else {
+            errorHandler(
+                Error(`Args callback did not return an array:` + inspect(args))
```

Comment:
We're using a template literal here without any interpolated values, and then string concat. Should choose one or the other